shared/src/androidTest/kotlin/co/touchlab/kmp/PlatformTest.kt

```diff
@@ -1,13 +1,52 @@
 package co.touchlab.kampstarter
 
 import androidx.test.ext.junit.runners.AndroidJUnit4
-import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.Rule
+import org.junit.rules.TestWatcher
+import org.junit.runner.Description
 import org.junit.runner.RunWith
 
-internal actual fun <T> runTest(block: suspend () -> T) { runBlocking { block() } }
+internal actual fun <T> runTest(block: suspend CoroutineScope.() -> T) {
+    runBlocking {
+        block()
+    }
+}
+
+/**
+ * Use this rule to update the Main dispatcher ahead of tests. By delegating the main dispatcher to a new thread.
+ * we can block the current thread and still dispatch main coroutines
+ */
+class CoroutineTestRule(
```

Comment:
I thought there was a builtin for this but maybe not

Reply:
I can't find anything. Feel like it would be mentioned here if so. https://github.com/Kotlin/kotlinx.coroutines/tree/master/kotlinx-coroutines-test#dispatchersmain-delegation

Reply:
Yeah I must be remembering wrong